CodingTracker.DreamFXX/DatabaseManager.cs

```diff
@@ -0,0 +1,140 @@
+
+        public void CreateDatabase()
+        {
+            using (var connection = new SqliteConnection(_connectionString))
+            {
+                connection.Open();
+                connection.Execute(@"CREATE TABLE IF NOT EXISTS MyCodingTracker (
+                                        Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                                        Date TEXT,
+                                        StartTime TEXT,
+                                        EndTime TEXT,
+                                        Duration TEXT)");
+            }
+        }
+
+        public void InsertRecord(string? date, string? startTime, string? endTime, string duration)
+        {
+            using (var connection = new SqliteConnection(_connectionString))
+            {
+                connection.Open();
+                connection.Execute(@$"INSERT INTO MyCodingTracker (Date, StartTime, EndTime, Duration)
+                                       VALUES('{date}', '{startTime}', '{endTime}', '{duration}')");
+            }
+        }
+
+        public void DeleteRecord(string? date)
+        {
+            using (var connection = new SqliteConnection(_connectionString))
+            {
+                connection.Open();
+                string sql = @$"DELETE FROM MyCodingTracker
+                                            WHERE Date = '{date}'";
+                connection.Execute(sql);
+            }
+        }
+
+        public void UpdateRecord(string? oldDate, string? newDate, string? startTime, string? endTime, string? duration)
+        {
+            using (var connection = new SqliteConnection(_connectionString))
+            {
+                connection.Open();
+                
+                    if (startTime is null && endTime is null)
+                    {
+                        connection.Execute($@"UPDATE MyCodingTracker
+                                                SET Date = '{newDate}'
+                                                WHERE Date = '{oldDate}'");
+                    }
+                    else if (newDate is null)
+                    {
+                        connection.Execute($@"UPDATE MyCodingTracker
+                                                SET StartTime = '{startTime}', EndTime = '{endTime}', Duration = '{duration}'
+                                                WHERE Date = '{oldDate}'");
+                    }
+                    else
+                    {
+                        connection.Execute($@"UPDATE MyCodingTracker
+                                               SET Date = '{newDate}', StartTime = '{startTime}', EndTime = '{endTime}', Duration = '{duration}'
+                                               WHERE Date = '{oldDate}'");
+                    }
+            }
+        }
+
+        public List<CodingSession> ReadFromDb()
```

Comment:
🔴 **ADO.NET**

Can you change to use Dapper instead of ADO.NET?
https://www.learndapper.com/dapper-query